src/NewButton/types.ts

```diff
@@ -27,10 +31,28 @@ export type ButtonProps = {
    * The trailing icon comes after button content
    */
   trailingIcon?: React.FunctionComponent<IconProps>
+  children: React.ReactNode
+  as?: React.ElementType
+} & ButtonBaseProps
+
+export type IconButtonProps = {
   /**
-   * Items that are disabled can not be clicked, selected, or navigated through.
+   * This is to be used if it is an icon-only button. Will make text visually hidden
    */
-  disabled?: boolean
-  children: React.ReactNode
-} & SxProp &
-  HTMLAttributes<HTMLButtonElement>
+  icon: React.FunctionComponent<IconProps>
+  iconLabel: string
+} & ButtonBaseProps
+
+// adopted from React.AnchorHTMLAttributes
+export type LinkButtonProps = {
+  underline?: boolean
```

Comment:
[help-wanted[: accessibility audit mentions that alink should be underlined on focus. Not sure how important it is.